### PR TITLE
INT: make "inject language intention" work

### PR DESCRIPTION
**Pull request by vlad20012**
Allow to inject any language to any string literal via intention

### CODE REVIEW

File: src/test/kotlin/org/rust/ide/intentions/InjectLanguageIntentionTest.kt [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        InlineFile(code.trimIndent()).withCaret()
+        val language = InjectedLanguage.findLanguageById(lang)
+        val injectable = Injectable.fromLanguage(language)
+        InjectLanguageAction.invokeImpl(project, myFixture.editor, myFixture.file, injectable)
+        val host = findInjectionHost(myFixture.editor, myFixture.file)
+        check(host != null)
+        val injectedList = InjectedLanguageManager.getInstance(project).getInjectedPsiFiles(host!!)
+        check(injectedList != null && injectedList.size == 1)
+        val injectedPsi = injectedList!!.first().first
+        check(injectedPsi.language == language)
```
Comment (Undin) [inline src/test/kotlin/org/rust/ide/intentions/InjectLanguageIntentionTest.kt]: Maybe use `assertEquals(language, injectedPsi.language)`? - it should produce more pretty error message

File: src/test/kotlin/org/rust/ide/intentions/InjectLanguageIntentionTest.kt [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+    private fun doTest(lang: String, @Language("Rust") code: String) {
+        InlineFile(code.trimIndent()).withCaret()
+        val language = InjectedLanguage.findLanguageById(lang)
+        val injectable = Injectable.fromLanguage(language)
+        InjectLanguageAction.invokeImpl(project, myFixture.editor, myFixture.file, injectable)
+        val host = findInjectionHost(myFixture.editor, myFixture.file)
+        check(host != null)
+        val injectedList = InjectedLanguageManager.getInstance(project).getInjectedPsiFiles(host!!)
+        check(injectedList != null && injectedList.size == 1)
```
Comment (Undin) [inline src/test/kotlin/org/rust/ide/intentions/InjectLanguageIntentionTest.kt]: Same comment about check for null

File: src/main/kotlin/org/rust/ide/injected/RsStringLiteralEscaper.kt [hunk 3]
```diff
@@ -52,7 +53,7 @@ private class RsRawStringLiteralEscaper(host: RsLitExpr) : LiteralTextEscaper<Rs
 }
 
 fun escaperForLiteral(lit: RsLitExpr): LiteralTextEscaper<RsLitExpr> {
-    val elementType = lit.firstChild?.elementType
+    val elementType = lit.node.findChildByType(RS_LITERALS)?.elementType
```
Comment (Undin) [inline src/main/kotlin/org/rust/ide/injected/RsStringLiteralEscaper.kt]: Hm, looks like I didn't answer it in the corresponding review.
How do we ensure that we call `escaperForLiteral` only for string literals?

File: src/test/kotlin/org/rust/ide/intentions/InjectLanguageIntentionTest.kt [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+import com.intellij.psi.PsiFile
+import com.intellij.psi.PsiLanguageInjectionHost
+import com.intellij.psi.injection.Injectable
+import com.intellij.psi.util.PsiTreeUtil
+import org.intellij.lang.annotations.Language
+import org.intellij.plugins.intelliLang.inject.InjectLanguageAction
+import org.intellij.plugins.intelliLang.inject.InjectedLanguage
+
+class InjectLanguageIntentionTest : RsIntentionTestBase(InjectLanguageAction()) {
+    private fun checkAvailable(@Language("Rust") code: String) {
```
Comment (Undin) [inline src/test/kotlin/org/rust/ide/intentions/InjectLanguageIntentionTest.kt]: Same suggestion as [here](https://github.com/intellij-rust/intellij-rust/pull/2683#discussion_r204214613)

**Comment by vlad20012** [comment]
Resolved

**Comment by vlad20012** [comment]
There are some problems with linking "IntelliLang" plugin, but it is needed for the tests only, so I moved the tests to another PR #2735. This is mergeable now

**Comment by vlad20012** [comment]
bors r=undin

**Pull request merged**